File: java/src/main/java/com/example/goban/Goban.java
package com.example.goban;

import java.util.List;

public class Goban {
    private final List<String> goban;

    public Goban(List<String> goban) {
        this.goban = goban;
    }

    public Status getStatus(int x, int y) {
        if (goban == null || goban.isEmpty() || x < 0 || y < 0 || y >= goban.size() || x >= goban.getFirst().length()) {
            return Status.OUT;
        }
        char stone = goban.get(y).charAt(x);
        return switch (stone) {
            case '.' -> Status.EMPTY;
            case 'o' -> Status.WHITE;
            case '#' -> Status.BLACK;
            default -> throw new IllegalArgumentException("Unknown goban value " + stone);
        };
    }

    public boolean isTaken(int x, int y) {
        throw new UnsupportedOperationException("Not implemented yet");
    }
}

File: php/src/Goban.php
<?php

declare(strict_types=1);

namespace Goban;

class Goban
{
    /** @var list<list<string>> */
    private array $grid;

    /** @param list<string> $lines */
    public function __construct(array $lines)
    {
        $this->grid = array_map(str_split(...), $lines);
    }

    public function getStatus(int $x, int $y): Status
    {
        if (empty($this->grid)) {
            return Status::OUT;
        }

        if ($x < 0 || $y < 0 || $y >= count($this->grid) || $x >= count($this->grid[0])) {
            return Status::OUT;
        }

        return match ($this->grid[$y][$x]) {
            '#' => Status::BLACK,
            'o' => Status::WHITE,
            '.' => Status::EMPTY,
        };
    }

    public function isTaken(int $x, int $y): bool
    {
        throw new \RuntimeException('Not implemented');
    }
}
